<launch>

  <node name="jetbot_msg_puller"  pkg="jetbot"  type="jetbot_msg_puller" output="screen">
  <param name="serial_port"         type="string" value="/dev/ttyTHS2"/>  
  <param name="serial_baudrate"     type="int"    value="115200"/>
  <param name="spin_rate"           type="int"    value="100"/>
  </node>

  <node name="jetbot_msg_pusher"  pkg="jetbot"  type="jetbot_msg_pusher" output="screen">
  <param name="serial_port"         type="string" value="/dev/ttyTHS2"/>  
  <param name="serial_baudrate"     type="int"    value="115200"/>
  <param name="spin_rate"           type="int"    value="50"/>
  </node>

  <node name="jetbot_odo_pub"  pkg="jetbot"  type="jetbot_odo_pub" output="screen">
  <param name="spin_rate"           type="int"    value="50"/>
  </node>

  <node name="jetbot_cmd_vel"  pkg="jetbot"  type="jetbot_cmd_vel" output="screen">
  <param name="spin_rate"           type="int"    value="50"/>
  <remap from="jetbot_cmd_vel/cmd_vel" to="/cmd_vel" />
  </node>

  <node name="jetbot_vel_repub"  pkg="jetbot"  type="jetbot_vel_repub" output="screen">
  <param name="spin_rate"           type="int"    value="50"/>
  <param name="desire_rate"           type="int"    value="20"/>
  <remap from="jetbot_cmd_vel/cmd_vel" to="cmd_vel" />
  <remap from="jetbot_vel_repub/cmd_vel" to="cmd_vel" />
  </node>

<!-- To save computation resource
  <node pkg="tf" type="static_transform_publisher" name="uwb_tf_broadcaster" args="0.26 0 0.4 0 0 0 1 base_link uwb 100" />  
-->

  <node name="jetbot_uwb_pose"  pkg="jetbot"  type="jetbot_uwb_pose" output="screen">
  <param name="spin_rate"           type="int"    value="50"/>
  </node>

  <node name="jetbot_relocator"  pkg="jetbot"  type="jetbot_relocator" output="screen">
  <param name="spin_rate"           type="int"    value="10"/>
  </node>

  <node name="jetbot_virtual_rc"  pkg="jetbot"  type="jetbot_virtual_rc" output="screen">
  <param name="spin_rate"           type="int"    value="50"/>
  </node>


</launch>
